app/src/main/java/org/simple/clinic/textInputdatepicker/TextInputDatePickerSheet.kt

```diff
@@ -20,6 +23,21 @@ class TextInputDatePickerSheet : BaseBottomSheet<
 
   override fun bindView(inflater: LayoutInflater, container: ViewGroup?) = SheetTextInputDatePickerBinding.inflate(inflater, container, false)
 
+  override fun events() = sheetCloseClicks()
+      .compose(ReportAnalyticsEvents())
+      .cast<TextInputDatePickerEvent>()
+
+  private val imageTextInputSheetClose
+    get() = binding.imageTextInputSheetClose
+
+  private fun sheetCloseClicks() = imageTextInputSheetClose
+      .clicks()
+      .map { DismissSheetClicked }
+
+  override fun dismissSheet() {
+    dismiss()
```

Comment:
Currently, dismiss doesn't remove the sheet from the back stack. Instead, we have to use `router.pop()`